gax-grpc/src/main/java/com/google/api/gax/grpc/OperationFuture.java

```diff
@@ -131,7 +175,34 @@ public String getName() throws ExecutionException, InterruptedException {
     }
   }
 
-  public Future<MetadataT> getMetadata() {
+  /**
+   * Gets the metadata of the operation tracked by this {@link OperationFuture}. This method returns
+   * the current poll metadata result (or the initial call metadata if it hasn't completed yet). The
+   * returned future completes once the current scheduled poll request (or the initial request if it
+   * hasn't completed yet) is executed and response is received from the server. The time when the
+   * polling request is executed is determined by the underlying polling algorithm.
+   *
+   * <p>Adding direct executor (same thread) callbacks to the future returned by this method is
+   * strongly not recommended, since the future is resolved under retrying future's internal lock
+   * and may affect the operation polling process. Adding separate thread callbacks is ok.
+   *
+   * <p>In most cases this method returns a future which is not completed yet, so calling {@link
+   * ApiFuture#get()} is a potentially blocking operation. To get metadata without blocking the
+   * current thread use the {@link #peekMetadata()} method instead.
+   *
+   * <p>In general this method behaves similarly to {@link RetryingFuture#getAttemptResult()}.
```

Comment:
Agree. I moved that line at the end of the documentation block, so only a very meticulous reader will ever read that line =). I still want to keep it there to provide a deeper look on how the method behaves. Even though conceptually it is easy to understand what the method is about, the way how it behaves has many pitfalls. There are many corner cases, possible outcomes and execution ordering combinations. It would be nice if a user, who got "weird" behavior, to get the explanation directly from the documentation without digging too much into implementation details. 

The main purpose of the link to RetryingFuture here is to document how and in which order attempts/attempt results/the whole future are resolved.